File: VMSFrontend/src/app/core/city.models.ts
/** The city master (FSD §16, §48.3 screen 8) — its own Trips-module entity, richer than the platform's generic
 * `CITY` lookup (name/abbreviation/province) which every other picker (`vms-lookup-picker type="CITY"`) reads. */
export interface CityModel {
  cityId: number;
  cityName: string;
  abbreviation: string;
  countryId: number;
  provinceState?: string | null;
  status: 'Active' | 'Inactive';
}

export interface SaveCityRequest {
  cityName: string;
  abbreviation: string;
  countryId?: number | null;
  provinceState?: string | null;
  status: 'Active' | 'Inactive';
}

File: src/VMS.Modules.Trips/Models/CityModels.cs
namespace VMS.Modules.Trips.Models;

public sealed class CityModel
{
    public int CityId { get; set; }
    public string CityName { get; set; } = string.Empty;
    public string Abbreviation { get; set; } = string.Empty;
    public int CountryId { get; set; }
    public string? ProvinceState { get; set; }
    public string Status { get; set; } = string.Empty;
    /// <summary>What the UI always shows (§16): <c>CityName (ABBR)</c>.</summary>
    public string Display => $"{CityName} ({Abbreviation})";
}

public sealed class SaveCityRequest
{
    public string CityName { get; set; } = string.Empty;
    public string Abbreviation { get; set; } = string.Empty;
    public int? CountryId { get; set; }
    public string? ProvinceState { get; set; }
    public string Status { get; set; } = Domain.ActiveInactiveStatuses.Active;
}
